src/ui/public/kuery/lib/filter_migration/index.js

```diff
@@ -1,21 +1 @@
-import { nodeTypes } from '../node_types';
-import { convertPhraseFilter } from './lib/phrase';
-import { convertRangeFilter } from './lib/range';
-import { convertExistsFilter } from './lib/exists';
-import { convertGeoBoundingBox } from './lib/geo_bounding_box';
-import { convertGeoPolygon } from './lib/geo_polygon';
-
-export function filterToKueryAST(filter) {
-  const { negate } = filter.meta;
-  const node = convertPhraseFilter(filter)
-    || convertRangeFilter(filter)
-    || convertExistsFilter(filter)
-    || convertGeoBoundingBox(filter)
-    || convertGeoPolygon(filter);
-
-  if (!node) {
-    throw new Error(`Couldn't convert that filter to a kuery`);
-  }
-
-  return negate ? nodeTypes.function.buildNode('not', node) : node;
-}
+export { filterToKueryAST } from './lib/filter_to_kuery';
```

Comment:
I feel like shoving everything into the top level `public` directory makes it very difficult to understand the most important relationships between modules. For example, ask a newcomer to infer the relationships of all the vislib code scattered throughout `public` and they're gonna have a bad time.

Let me propose a couple options that cut down on the `/lib/lib/lib`ness while still grouping all the kuery code in one spot. 

Option 1

![screen shot 2017-07-10 at 11 03 50 am](https://user-images.githubusercontent.com/6239176/28024736-8a5396ba-655f-11e7-88a7-269d8095c823.png)

Option 2

![screen shot 2017-07-10 at 11 00 46 am](https://user-images.githubusercontent.com/6239176/28024597-2e1947f0-655f-11e7-8e2e-5b9355225f8c.png)